File: Advanced DSA/01. Time Complexity/03. Different Bits Sum Pairwise/Solution.py
class Solution:
	# @param A : list of integers
	# @return an integer
	def cntBits(self, A):
        res, mod = 0, 1e9 + 7
        for i in range(32):
            set_bit_count = 0
            for j in range(len(A)):
                set_bit_count += bool((A[j] & (1 << i)) > 0)
            res = (res % mod) + (((set_bit_count * (len(A) - set_bit_count)) << 1) % mod)
        return int(res)

File: Advanced DSA/01. Time Complexity/03. Different Bits Sum Pairwise/Solution.cpp
int Solution::cntBits(vector<int> &A) {
    long res = 0, mod = 1e9 + 7;
    for (int i = 0; i < 32; ++i) {
        int set_bit_count = 0;
        for (int j = 0; j < A.size(); ++j)
            set_bit_count += bool(A[j] & (1 << i));
        res = (res % mod) + (((set_bit_count * (A.size() - set_bit_count)) << 1) % mod);
    }
    return res;
}
